li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -90,10 +104,19 @@ private LiKafkaFederatedConsumerImpl(LiKafkaConsumerConfig configs, MetadataServ
     // Each per-cluster consumer and auditor will be instantiated by the passed-in consumer builder when the client
     // begins to consume from that cluster. If a null builder is passed, create a default one, which builds
     // LiKafkaConsumer.
-    _consumers = new ConcurrentHashMap<ClusterDescriptor, LiKafkaConsumer<K, V>>();
+
+    _consumers = new ConcurrentSkipListMap<ClusterDescriptor, LiKafkaConsumer<K, V>>(
+        new Comparator<ClusterDescriptor>() {
+          public int compare(ClusterDescriptor cluster1, ClusterDescriptor cluster2) {
+            return cluster1.toString().compareTo(cluster2.toString());
```

Comment:
(to me at least) toString() is for debugging purposes. i'd rather compare by something like the bootstrap url or zk url

Reply:
Used bootstrap url.